icu4c/source/i18n/units_converter.cpp

```diff
@@ -491,14 +491,43 @@ Convertibility U_I18N_API extractConvertibility(const MeasureUnitImpl &source,
 
 UnitsConverter::UnitsConverter(const MeasureUnitImpl &source, const MeasureUnitImpl &target,
                              const ConversionRates &ratesInfo, UErrorCode &status)
-    : conversionRate_(source.copy(status), target.copy(status)) {
-    if (source.complexity == UMeasureUnitComplexity::UMEASURE_UNIT_MIXED ||
-        target.complexity == UMeasureUnitComplexity::UMEASURE_UNIT_MIXED) {
+ {
+     if(U_FAILURE(status)) {
+         return;
+     }
+
+     this->conversionRate_.source = source.copy(status);
+     this->conversionRate_.target = target.copy(status);
```

Comment:
Here and below: It's a bit nicer and more efficient if you initialize fields using constructor initialization (like you had before, `: conversionRate(...) {`.

You can even put the whole MeasureUnitImpl thing here:

```
: conversionRate_(
    MeasureUnitImpl::forIdentifier(sourceIdentifier, status),
    MeasureUnitImpl::forIdentifier(targetIdentifier, status),
)
```

Reply:
done.